src/datachain/data_storage/warehouse.py

```diff
@@ -227,6 +228,9 @@ def dataset_select_paginated(
                         paginated_query = paginated_query.limit(None).limit(limit)
 
                 results = self.db.execute(paginated_query.offset(offset))
+                results = convert_rows_custom_column_types(
+                    query.selected_columns, results, self.db.dialect
+                )
```

Comment:
We already have `dataset_rows_select()` in SQLite and in CH, that do similar things. 
Maybe we can extract this out and unify them? That way, we will only have one place to `execute` and convert rows.

Reply:
🤔 I like the idea. Let me think.

Reply:
Please, check the update [here](https://github.com/iterative/datachain/pull/339/commits/ccd0572d7549bb2ff22036b1dbc5cdaa5f875061) and in related SaaS PR.

Reply:
Can we also change this above line to use `dataset_rows_select`?
```suggestion
                results = self.dataset_rows_select(paginated_query.offset(offset))
```